docs/howto.md

```diff
@@ -0,0 +1,174 @@
+    // We need io_context for IO, this is alias on boost::asio::io_context
+    boost::asio::io_context io;
+
+    // Rows which accepts integer and nullable string columns in the sequence
+    // It is an alias on std::vector of std::tuple
+    ozo::rows_of<std::int64_t, std::optional<std::string>> rows;
+
+    // Connection info with host and port to coonect to
+    ozo::connection_info<> conn_info("host=... port=...");
+
+    // Our query statement
+    const auto query = "SELECT id, name FROM users_info WHERE amount>="_SQL + std::int64_t(25);
+
+    // Request with connection provider, query and callback.
+    // Provider binds how to get connection with io_context.
+    ozo::request(ozo::make_provider(io, conn_info), query, ozo::into(rows),
+            [&](ozo::error_code ec, auto conn) {
+        if (ec) {
+            // Here we got an error, so we can get:
+            //           error code's message
+            std::cerr << ec.message()
+            //           error message from underlying libpq
+                    << " | " << error_message(conn)
+            //           and error context from OZO
+                    << " | " << get_error_context(conn);
+            return;
+        };
+
+        // Connection must be in good state here,
+        // typically you do not need to check it manually
+        assert(ozo::connection_good(conn));
+
+        // We got results, let's handle, e.g. print it out
+        std::cout << "id" << '\t' << "name" << std::endl;
+        for(auto& row: res) {
+            std::cout << std::get<0>(row) << '\t'
+                    << std::get<1>(row) << std::endl;
+        }
+    });
+
+    io.run();
+}
+```
+
+Let's look a little bit closer on this pretty simple asynchronous code.
+
+```c++
+ozo::rows_of<std::int64_t, std::optional<std::string>> rows;
+```
+
+Here we define result type as we want to see it. Practically `ozo::rows_of` is an alias on `std::vector<std::tuple<...>>`. And `ozo::into` is an alias on `std::back_inserter`. So `request()` function will fill this vector of tuples according to the database response rows. It is very important to preserve the same order of fields in request and types in the tuple (it is a little bit annoyng, but there is a way to avoid it via [Boost.Hana](https://www.boost.org/doc/libs/1_66_0/libs/hana/doc/html/index.html#tutorial-introspection-adapting) or [Boost.Fusion](https://www.boost.org/doc/libs/1_66_0/libs/fusion/doc/html/fusion/adapted.html) structure adaptation). We can see `std::optional` at second position in the tuple. The reason is what `name` field can be _NULL_. So the empty optional is the _NULL_ representation (see nullables for more info about it). If you ommit `std::optional` then in case of _NULL_ there will be a result deserialization error. The first argument can not be a _NULL_, so here we do not to bother about it. There is no mistake to expect nullable type for non-nullable result, but the opposite leads to run-time error.
```

Comment:
Split this paragraph by lines of ~100 symbols.

Reply:
`annoyng` -> `annoying`

Reply:
**Done**